source/extensions/filters/http/gcp_authn/gcp_authn_filter.cc

```diff
@@ -12,8 +14,43 @@ namespace HttpFilters {
 namespace GcpAuthn {
 
 using ::Envoy::Router::RouteConstSharedPtr;
+using ::google::jwt_verify::Status;
 using Http::FilterHeadersStatus;
 
+void addTokenToRequest(Http::RequestHeaderMap& hdrs, absl::string_view token_str) {
+  std::string id_token = absl::StrCat("Bearer ", token_str);
+  hdrs.addCopy(authorizationHeaderKey(), id_token);
+}
+
+template <typename TokenType> TokenType* TokenCacheImpl<TokenType>::lookUp(const std::string& key) {
+  ASSERT(lru_cache_ != nullptr);
+  typename LRUCache<TokenType>::ScopedLookup lookup(lru_cache_.get(), key);
+  if (lookup.found()) {
+    TokenType* const found_token = lookup.value();
+    if constexpr (std::is_same<TokenType, JwtToken>::value) {
```

Comment:
It is a good point to consider. But at this point, I feel cache implementation is a good/better place for`removing the expired` because:
1. removing is a natural part of LRU cache (like. eviction)
2. I am planning to add the functionality of recycling all expired entries either through setting [max idle time](https://source.corp.google.com/piper///depot/google3/third_party/jwt_verify_lib/simple_lru_cache/simple_lru_cache_inl.h;rcl=450333622;l=308) or periodically through [functor](https://source.corp.google.com/piper///depot/google3/third_party/jwt_verify_lib/simple_lru_cache/simple_lru_cache.h;rcl=387869691;l=42) , which will make the `removing` even more tightly coupled with cache. 

I have thought thoroughly about #2 yet. So how about we evaluate this in next PR? I moved remove part to a separate function to make it more readable.